### ISSUE TITLE
SEGV in comp_compare func

### ISSUE THREAD

**Issue by Jminis** [libical/libical#639]
Hello, I would like to share a null-deref crash that I discovered while independently running oss-fuzz.

Below are the fuzzer, target function information, input data, and ASAN log.

# Fuzz Information 
fuzzer binary : libical_extended_fuzzer
crash state : `comp_compare` in `/src/libical/src/libical/icalcomponent.c`
input data : ```\x79\x79\x42\x79\xBD\x0A\x3B\x4F\x4E\x45\x4E\x45\x3A\x2E\x0A\x47\x45\x3B\x4E\x42\x4E\x3B\x0A\x0A\x45\x0A\x0A\x42\x45\x47\x49\x4E\x3B\x0A\x42\x45\x15\x45\x3B\x0A\x42\x45\x47\x49\x4E\x3B\x56\x54\x49\x4D\x45\x5A\x4F\x4E\x45\x4E\x45\x3A\x2E\x0A\x47\x45\x3B\x4E\x42\x4E\x3B\x0A\x0A\x45\x4E\x44\x3B\x5C\x0A\x49\x0A\x0A\x36\x28\x0A\x0A\x3B\x0A\x45\x0A\x42\x45\x47\x54\x45\x4D\x56\x49\x4E\x49\x3B\x5A\x4F\x4E\x3B\x0A\x0A\x42\xB5\x47\x49\x4E\x3B\x56\x54\x49\x4D\x45\x5A\x4F\x4E\x45\x4E\x45\x3A\x6E\x0A\x49\x4E\x42\x45\x47\x3B\x0A\xF6\xF5\xD7\xF9\x0A\x30\x3B\x0A\x45\x42\x45\x15\x45\x3B\x0A\x42\x45\x47\x49\x4E\x3B\x56\x54\x49\x4D\x45\x5A\x4F\x4E\x45\x4E\x45\x3A\x2E\x0A\x47\x45\x3B\x4E\x42\x4E\x3B\x0A\x0A\x45\x4E\x44\x3B\x5C\x0A\x49\x4E\x44\xF4\x0A\x45\x4E\x44\x3A\x2E\x0A\x49\x4E\x0A\x42\x45\x47\xDD\xAC\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xAB\x09\xF1\x00\x00\x44\x3B\x4A\x00\x36\x0D\x0D```

```
AddressSanitizer:DEADLYSIGNAL
=================================================================
==11==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000000 (pc 0x0000004c46b0 bp 0x7ffdfa52d090 sp 0x7ffdfa52c840 T0)
==11==The signal is caused by a READ memory access.
==11==Hint: address points to the zero page.
SCARINESS: 10 (null-deref)
    #0 0x4c46b0 in strcmp /src/llvm-project/compiler-rt/lib/asan/../sanitizer_common/sanitizer_common_interceptors.inc:461:25
    #1 0x57ecc9 in comp_compare /src/libical/src/libical/icalcomponent.c
    #2 0x5c6b2c in pvl_insert_ordered /src/libical/src/libical/pvl.c:250:10
    #3 0x57e2b7 in icalcomponent_normalize /src/libical/src/libical/icalcomponent.c:2695:9
    #4 0x56c465 in LLVMFuzzerTestOneInput /src/libical_extended_fuzzer.cc:40:9
    #5 0x43de23 in fuzzer::Fuzzer::ExecuteCallback(unsigned char const*, unsigned long) /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerLoop.cpp:611:15
    #6 0x429582 in fuzzer::RunOneTest(fuzzer::Fuzzer*, char const*, unsigned long) /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerDriver.cpp:324:6
    #7 0x42ee2c in fuzzer::FuzzerDriver(int*, char***, int (*)(unsigned char const*, unsigned long)) /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerDriver.cpp:860:9
    #8 0x458362 in main /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerMain.cpp:20:10
    #9 0x7f7185102082 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x24082) (BuildId: 1878e6b475720c7c51969e69ab2d276fae6d1dee)
    #10 0x41f74d in _start (/out/libical_extended_fuzzer+0x41f74d)

DEDUP_TOKEN: strcmp--comp_compare--pvl_insert_ordered
AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV /src/llvm-project/compiler-rt/lib/asan/../sanitizer_common/sanitizer_common_interceptors.inc:461:25 in strcmp
```

After conducting a brief debugging, it was found that the argument passed to strcmp within the comp_compare function was NULL.

# Analysis
1. Debugging just before the SEGV occurred in the strcmp function
- It can be confirmed that both the first and second arguments of the strcmp function are NULL.

![image](https://user-images.githubusercontent.com/68144858/235609748-5479d6cd-bf0b-48fb-9691-f957370a4599.png)

2. Debugging just before the argument to strcmp becomes NULL
- The return value of the icalcomponent_get_first_property function was NULL, causing the icalproperty_get_value_as_string return value to also be NULL.

![image](https://user-images.githubusercontent.com/68144858/235609738-f4fb5db8-e533-4d31-aef4-62080e83dfe8.png)

**Comment by winterz** [comment]
@Jminis I believe this one has been fixed recently (by 8a04e0b7c98301da1ef60549fd4a977b8aa7b970 I think)

can you check?

**Comment by Jminis** [comment]
@winterz winterz Thank you for reviewing my comment.

I have personally confirmed that the issue I previously raised does not occur in the latest version of libical!

![image](https://github.com/libical/libical/assets/68144858/8fbbfe2c-9389-4e57-9e2f-6754cecff3d1)
